File: SphrLibrary/Entities/OpenmHealth/PhysicalActivityBody.cs
using System.Runtime.Serialization;

namespace SphrLibrary.Entities.OpenmHealth
{
    /// <summary>
    /// Open mHealth 歩数データボディ部のリソースを表します。
    /// </summary>
    [DataContract()]
    [Serializable()]
    public class PhysicalActivityBody : OmhBodyEntityBase
    {
        [DataMember()]
        public string activity_name = string.Empty;
        //[DataMember()]
        //public EffectiveTimeFrame effective_time_frame = new EffectiveTimeFrame();
        [DataMember()]
        public ValueUnit distance = new ValueUnit();
        [DataMember()]
        public ValueUnit base_movement_quantity = new ValueUnit();

        public PhysicalActivityBody() : base() { }
    }
}

File: References/Java/SphrLibrary-Java/lib/src/main/java/sphrlibrary/entities/openmhealth/PhysicalActivityBody.java
package sphrlibrary.entities.openmhealth;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.ToString;

/**
 * 歩数データの Body
 */
@Data
@NoArgsConstructor
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class PhysicalActivityBody extends OmhBodyEntityBase {

    /**
     * その人が従事している身体活動の名称。
     */
    @NonNull
    private String activityName = "";

    /**
     * 距離
     */
    @NonNull
    private ValueUnit distance = new ValueUnit();

    /**
     * 身体活動の基本動作の繰り返し回数。活動が歩行の場合、base_movement_quantityは歩数となる。
     */
    @NonNull
    private ValueUnit baseMovementQuantity = new ValueUnit();
}
